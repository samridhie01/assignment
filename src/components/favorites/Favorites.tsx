import React, {useState, useEffect} from "react";
import PlanetTile from "../planets/PlanetTile";
import {Box, Button, makeStyles} from "@material-ui/core";
import Header from "../header/Header";
import isEmpty from "lodash/isEmpty";
import {PlanetData} from "../types/types";


const useStyles = makeStyles({
    alignContent: {
        alignSelf: "center",
    }
})

const Favorites = () => {
    const [favs, setFavs] = useState<PlanetData[]>([]);

    useEffect(() => {
        setFavs(JSON.parse(localStorage.getItem("favs") || "{}"));
    }, [])


    const handleRemoveFavs = (favItem: string) => {
        const addedFavs = JSON.parse(localStorage.getItem("favs")!);
        const newList = addedFavs.filter((data1: PlanetData) => {
            return data1.name !== favItem;
        });
        if (isEmpty(newList)) {
            localStorage.setItem("favs", "");
            setFavs([])
        } else {
            localStorage.setItem("favs", JSON.stringify(newList));
            setFavs(newList)
        }

    }

    const classes = useStyles();
    return (
        <div data-testid={"favorties-section"}>
            <Header path="/" linkName={"HomePage"}/>
            {isEmpty(favs) ? (
                <h2>NO DATA TO SHOW</h2>
            ) : (
                {favs} &&
                favs.map(data => {
                    return (
                        <Box display="flex" flexDirection="row" justifyContent="space-around" m={2} border={1}
                             width={800}>
                            <PlanetTile name={data.name} climate={data.climate}/>
                            <Button variant="contained" size="small" className={classes.alignContent} onClick={() => {
                                handleRemoveFavs(data.name)
                            }}>Remove</Button>
                        </Box>

                    );
                })
            )}
        </div>
    );
};

export default Favorites;
