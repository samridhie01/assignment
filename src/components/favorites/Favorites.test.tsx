import {render, fireEvent, getByText, getByTestId, wait} from "@testing-library/react";

import React from "react";
import Favorites from "./Favorites";
import {BrowserRouter} from "react-router-dom";


describe("Favorites Page", ()=>{

    localStorage.setItem("favs",JSON.stringify([{"name":"abc", "climate":"def"}]));
    const renderComponent = ()=>
        render(
            <BrowserRouter>
                <Favorites/>
            </BrowserRouter>)


    it("renders screenshot", ()=>{
        const { asFragment } = render(
            <BrowserRouter>
            <Favorites />
             </BrowserRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("validate removal of favorites item", ()=>{
        const {getByText} = renderComponent();
        const removeBtn = getByText("Remove");
        expect(removeBtn).toBeInTheDocument();
        fireEvent.click(removeBtn);
        expect(localStorage.getItem("favs")).toEqual("");
        expect(getByText("NO DATA TO SHOW"))
    })
})
