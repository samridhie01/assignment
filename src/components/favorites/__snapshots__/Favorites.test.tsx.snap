// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites Page renders screenshot 1`] = `
<DocumentFragment>
  <div>
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorInherit MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="menu"
          class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <img
              alt="website logo"
              src="planet2.svg"
              style="height: 100px; width: 100px;"
            />
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <h6
          class="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
        >
          Planets
        </h6>
        <a
          color="inherit"
          href="/"
        >
          HomePage
        </a>
      </div>
    </header>
    <h2>
      NO DATA TO SHOW
    </h2>
  </div>
</DocumentFragment>
`;
